addon/components/modal-dialog.js

```diff
@@ -117,6 +118,25 @@ export default Ember.Component.extend({
     }
   }),
 
+  didInsertElement() {
+    if (!this.get('clickOutsideToClose')) {
+      return;
+    }
+
+    var handleClick = event => {
+      if (!$(event.target).closest('.ember-modal-dialog').length) {
+        this.send('close');
+      }
+    };
+    var registerClick = () => $(document).on('click.ember-modal-dialog', handleClick);
+
+    // setTimeout needed or else the click handler with catch this current action
```

Comment:
Comment should probably read `will catch the click that spawned this modal dialog`